core/src/main/java/com/linecorp/armeria/server/RoutingTrie.java

```diff
@@ -223,16 +223,27 @@ private Node<V> checkNode(Node<V> node, String path, int begin, boolean exact, I
                 break;
             case PARAMETER:
                 // Consume characters until the delimiter '/' as a path variable.
-                final int delim = path.indexOf('/', begin);
-                if (delim < 0) {
-                    // No more delimiter.
-                    return node;
+                final int delimSlash = path.indexOf('/', begin);
+
+                if (delimSlash < 0) {
+                    final int lastColon = path.lastIndexOf(':');
+                    // lastIndexOf performs backward search
+                    if (lastColon < begin) {
+                        // No more delimiter.
+                        return node;
+                    } else {
+                        final Node<V> verb = node.children.get(':');
+                        final String pathVerbPart = path.substring(lastColon);
```

Comment:
Micro optimization) What do you think of implementing a method such as `findVerb` to find a verb part from `path`? `findVerb` will scan `path` from the last character up to `begin` to find `:`.
```java
String verbPart = findVerb(path, begin);
if (verbPart == null) {
  return node;
} else {
   final Node<V> verb = node.children.get(':');
   ...
}

```

Reply:
I think we should use the `fineVerb` method defined in `VerbSuffixPathMapping` because the verb found by the method has to be last segment of a path 👍

Reply:
The regular expression may consume more CPU cycles which is inefficient in finding an exact character.

Reply:
Thank you for your advice! Implemented the `findVerb` function.